src/decrypters/widevine/WVCencSingleSampleDecrypter.cpp

```diff
@@ -514,123 +540,99 @@ AP4_Result CWVCencSingleSampleDecrypter::DecryptSampleData(AP4_UI32 poolId,
+                         subsOut * sizeof(AP4_UI16));
+      dataOut.AppendData(reinterpret_cast<const AP4_Byte*>(rebuiltBytesOfEncryptedData.data()),
+                         subsOut * sizeof(AP4_UI32));
       dataOut.AppendData(reinterpret_cast<const AP4_Byte*>(iv), 16);
       dataOut.AppendData(fragInfo.m_key.data(), static_cast<AP4_Size>(fragInfo.m_key.size()));
     }
     else
     {
       dataOut.SetDataSize(0);
-      bytesOfCleartextData = &dummyClear;
-      bytesOfEncryptedData = &dummyCipher;
     }
 
-    if (fragInfo.m_nalLengthSize && (!iv || bytesOfCleartextData[0] > 0))
-    {
-      //check NAL / subsample
-      const AP4_Byte *packetIn(dataIn.GetData()),
-          *packetInEnd(dataIn.GetData() + dataIn.GetDataSize());
-      // Byte position of "bytesOfCleartextData" where to set the size of the data,
-      // by default starts after the subsample count data (so the size of subsampleCount data type)
-      size_t clrDataBytePos = sizeof(subsampleCount);
-      // size_t nalUnitCount = 0; // is there a use for this?
-      size_t nalUnitSum = 0;
-      // size_t configSize = 0; // is there a use for this?
-
-      while (packetIn < packetInEnd)
-      {
-        uint32_t nalSize(0);
-        for (size_t i = 0; i < fragInfo.m_nalLengthSize; ++i)
-        {
-          nalSize = (nalSize << 8) + *packetIn++;
-        };
+#ifdef TARGET_WEBOS
+    // Prepare CDM input using rebuiltSubs
+    cdm::InputBuffer_2 cdmIn;
+    SetInput(cdmIn, payload, static_cast<uint32_t>(rebuiltSubs.size()), iv, fragInfo, rebuiltSubs);
 
-        //look if we have to inject sps / pps
-        if (!fragInfo.m_annexbSpsPps.empty() && (*packetIn & 0x1F) != 9 /*AVC_NAL_AUD*/)
-        {
-          dataOut.AppendData(fragInfo.m_annexbSpsPps.data(),
-                             static_cast<AP4_Size>(fragInfo.m_annexbSpsPps.size()));
-          if (iv)
-          {
-            // Update the byte containing the data size of current subsample referred to clear bytes array
-            AP4_UI16* clrb_out = reinterpret_cast<AP4_UI16*>(dataOut.UseData() + clrDataBytePos);
-            *clrb_out += fragInfo.m_annexbSpsPps.size();
-          }
-
-          // configSize = fragInfo.m_annexbSpsPps.GetDataSize();
-          fragInfo.m_annexbSpsPps.clear();
-        }
-        // Annex-B Start pos
-        static AP4_Byte annexbStartCode[4] = {0x00, 0x00, 0x00, 0x01};
-        dataOut.AppendData(annexbStartCode, 4);
-        dataOut.AppendData(packetIn, nalSize);
-        packetIn += nalSize;
-        
-        if (iv)
-        {
-          // Update the byte containing the data size of current subsample referred to clear bytes array
-          AP4_UI16* clrb_out = reinterpret_cast<AP4_UI16*>(dataOut.UseData() + clrDataBytePos);
-          *clrb_out += (4 - fragInfo.m_nalLengthSize);
-        }
-        
-        // ++nalUnitCount;
+    const size_t header_cushion = 64;
+    m_decryptOut.Reserve(payload.GetDataSize() + dataOut.GetDataSize() + header_cushion);
```

Comment:
have you some info about this `header_cushion`, cushion?
if so, please choose a better var name,
and move this inside an unnamed namespace on top by using constexpr, or otherwise at least should be `static const`